libs/blocks/marquee/marquee.js

```diff
@@ -42,13 +40,16 @@ const decorateBlockBg = (block, node) => {
     if (childCount === 3) {
       child.classList.add(viewports[index]);
     }
-
-    if (child.querySelector('a[href$=".mp4"]')) {
-      decorateVideo(child);
+    const videoLink = child.querySelector('a[href$=".mp4"]');
+    const videoElement = child.querySelector('video source[src$=".mp4"]');
+    if (videoLink) {
+      decorateVideo(child, videoLink.href);
+    } else if (videoElement) {
+      decorateVideo(child, videoElement.src);
     }
```

Comment:
Not a huge fan of else if's if they can be avoided:
```suggestion
const videoElement = child.querySelector('a[href$=".mp4"], video source[src$=".mp4"]');
if (videoElement) {
    decorateVideo(child, videoElement.href || videoElement.src);
}
```